### PR TITLE
Common Crypt4GH functionality

### PR DESCRIPTION
**Pull request by mephenor**
### Changes

- Collected Crypt4GH functionality used across file services and improved ergonomics a bit
- Changed `big_temp_file` to deal with `sys.set_int_max_str_digits` instead of delegating it to the caller
- Further repo structure changes and dependency updates to align better with microservice template making this compatible with the current `ghga-de/gh-action-common`

If I've overdone it with the adaptation to the template, please specify which changes I should roll back

### CODE REVIEW

File: .devcontainer/dev_install [hunk 1]
```diff
@@ -1,16 +1,16 @@
 #!/bin/bash
-# install service library in dev container
+# install service in dev container
```
Comment (mephenor) [inline .devcontainer/dev_install:2]:
```suggestion
# install service library in dev container
```

File: src/ghga_service_commons/utils/crypt4gh.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+def random_encrypted_content(
+    file_size: int, private_key: Union[str, bytes], public_key: Union[str, bytes]
+) -> RandomEncryptedData:
+    """Create an in-memory file with random content that is Crypt4GH encrypted."""
+    encrypted_file = io.BytesIO()
+
+    with big_temp_file(file_size) as raw_file:
+        # rewind input file for reading
+        raw_file.seek(0)
+        true_size = len(raw_file.read())
```
Comment (Cito) [inline src/ghga_service_commons/utils/crypt4gh.py:186]:
```suggestion
        true_size = raw_file.tell()
        # rewind input file for reading
        raw_file.seek(0)
```

should be a bit faster

**Review by Cito — CHANGES_REQUESTED** [review]
Thanks, looks good. Just a couple suggestions for further improvement.

File: src/ghga_service_commons/utils/crypt4gh.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    """Crypt4GH keypair"""
+
+    private: bytes
+    public: bytes
+
+
+class RandomEncryptedData(NamedTuple):
+    """Container for random Crypt4GH encrypted data.
+
+    User has to take care of closing the content object.
```
[the remark this answers is not in the document]
Comment (mephenor) [reply src/ghga_service_commons/utils/crypt4gh.py:51]: Yeah, forgot to update that after adding the context manager functionality...

File: src/ghga_service_commons/utils/crypt4gh.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        self.close()
+
+    def close(self):
+        """Close in-memory BytesIO content."""
+        self.content.close()
+
+
+def key_secret_decoder(function: Callable):
+    """Decorator decoding string arguments from base64 to bytes"""
+
```
[the remark this answers is not in the document]
Comment (mephenor) [reply src/ghga_service_commons/utils/crypt4gh.py:72]: Nice, somehow I did not know about this

File: src/ghga_service_commons/utils/crypt4gh.py (hunk 2 above)
Comment (mephenor) [reply src/ghga_service_commons/utils/crypt4gh.py:186]: Thanks for catching that one, just copied that from a service and did not take a second look

**Review by Cito — APPROVED** [review]

**Pull request merged**